pkg/southbound/synchronizer/synchronizer.go

```diff
@@ -44,12 +44,13 @@ type Synchronizer struct {
 	key                  southbound.DeviceID
 	query                client.Query
 	operationalCache     map[string]string
+	modelReadOnlyPaths   map[string][]string
 }
 
 // New Build a new Synchronizer given the parameters, starts the connection with the device and polls the capabilities
 func New(context context.Context, changeStore *store.ChangeStore, configStore *store.ConfigurationStore,
-	device *topocache.Device, deviceCfgChan <-chan events.ConfigEvent,
-	opStateChan chan<- events.OperationalStateEvent, errChan chan<- events.DeviceResponse) (*Synchronizer, error) {
+	device *topocache.Device, deviceCfgChan <-chan events.ConfigEvent, opStateChan chan<- events.OperationalStateEvent,
+	errChan chan<- events.DeviceResponse, mReadOnlyPaths map[string][]string) (*Synchronizer, error) {
```

Comment:
I think you only need to pass in an **[]string** - i.e. the one for the type of device you are creating